client/components/data/query-post-types/index.jsx

```diff
@@ -24,10 +27,19 @@ class QueryPostTypes extends Component {
 	}
 
 	componentWillReceiveProps( nextProps ) {
-		const { siteId, themeSlug } = this.props;
-		const { siteId: nextSiteId, themeSlug: nextThemeSlug } = nextProps;
+		const {
+			postTypeSettings,
+			siteId,
+			themeSlug
+		} = this.props;
+		const {
+			siteId: nextSiteId,
+			themeSlug: nextThemeSlug
+		} = nextProps;
 		const hasThemeChanged = themeSlug && nextThemeSlug && themeSlug !== nextThemeSlug;
-		if ( siteId !== nextSiteId || hasThemeChanged ) {
+		const hasPostTypeSettingChanged = ! isEqual( postTypeSettings, nextProps.postTypeSettings );
```

Comment:
A `nextPostTypeSettings` might be nice for symmetry, but not really a problem as it is.